ddtrace/tracer/sampler.go

```diff
@@ -147,3 +154,177 @@ func (ps *prioritySampler) apply(spn *span) {
 	}
 	spn.SetTag(keySamplingPriorityRate, rate)
 }
+
+type rulesSampler struct {
+	rules   []samplingRule
+	rate    float64
+	limiter *rate.Limiter
+	// "effective rate" calculations
+	mu           sync.Mutex
+	ts           int64
```

Comment:
ts is too vague. What does it mean?

Reply:
I think `ts` would make sense if the type would be `time.Time`, in this case maybe `timestamp` is better, or at least a comment to the right.

Reply:
Can you please make comments to the right of every field and describe what they are?

Reply:
Done